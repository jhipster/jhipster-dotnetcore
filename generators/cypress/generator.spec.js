import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'cypress';
const BLUEPRINT_NAMESPACE = `jhipster:${SUB_GENERATOR}`;

describe('SubGenerator cypress of dotnetcore JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(BLUEPRINT_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
          blueprint: 'dotnetcore',
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
