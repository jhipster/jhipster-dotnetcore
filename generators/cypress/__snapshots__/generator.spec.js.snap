// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator cypress of dotnetcore JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/cypress-audits.config.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/cypress.config.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/package.json": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/account/login-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/account/password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/account/register-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/account/reset-password-page.cy.ts": {
    "state": "deleted",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/account/settings-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/administration/administration.cy.ts": {
    "state": "deleted",
  },
  "src/Jhipster/ClientApp/test/cypress/e2e/lighthouse.audits.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/support/account.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "src/Jhipster/ClientApp/test/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;
